qa/zcash/updatecheck.py

```diff
@@ -112,24 +112,22 @@ def get_dependency_list():
 
     return dependencies
 
-class GitHubToken:
-    def __init__(self):
-        token_path = os.path.join(SOURCE_ROOT, ".updatecheck-token")
-        try:
-            with open(token_path, encoding='utf8') as f:
-                token = f.read().strip()
-                self._user = token.split(":")[0]
-                self._password = token.split(":")[1]
-        except:
-            print("Please make sure a GitHub API token is in .updatecheck-token in the root of this repository.")
-            print("The format is username:hex-token.")
-            sys.exit(1)
-
-    def user(self):
-        return self.user
-
-    def password(self):
-        return self.password
+def parse_token():
+    token_path = os.path.realpath(os.path.join(SOURCE_ROOT, ".updatecheck-token"))
+    if not os.path.exists(token_path):
+        token_path = os.path.join(xdg.xdg_data_home(), "zcash/updatecheck/token")
+    try:
+        with open(token_path, encoding='utf8') as f:
+            token = f.read().strip()
+            return token.split(":")[-1]
+    except:
+        print("You are missing a GitHub API token. This script will probably still work, but")
+        print("you are more likely to hit an API rate limit. Create a file named")
+        print(token_path)
+        print("containing the token to silence this warning.")
```

Comment:
```suggestion
        print("containing the token to silence this warning. The format is username:hex-token.")
```

Reply:
The format is no longer `username:hex-token`. That’ll still parse correctly if that’s what you have, but the change from `HTTPBasicAuth` means the username is no longer required.